File: scripts/export-bindings.mjs
import { spawnSync } from "node:child_process";
import { readFileSync, writeFileSync } from "node:fs";

if (!process.argv.includes("--normalize-only")) {
  const result = spawnSync(
    "cargo",
    [
      "run",
      "--no-default-features",
      "--manifest-path",
      "src-tauri/Cargo.toml",
      "--bin",
      "export-bindings",
    ],
    {
      env: { ...process.env, EXPORT_BINDINGS: "1" },
      stdio: "inherit",
      shell: process.platform === "win32",
    },
  );

  if (result.error) {
    throw result.error;
  }
  if (result.status !== 0) {
    process.exit(result.status ?? 1);
  }
}

const output = new URL("../src/generated/bindings.ts", import.meta.url);
const contents = readFileSync(output, "utf8");
writeFileSync(output, `${contents.trimEnd()}\n`);

File: src-tauri/src/bin/export-bindings.rs
use std::path::PathBuf;

fn main() {
    let output = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("../src/generated/bindings.ts");
    arknights_operation_runner_lib::export_bindings(output)
        .expect("生成 Tauri TypeScript 绑定失败");
}
